ci_framework/roles/ci_nmstate/tasks/nmstate_k8s_provision_node.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Prepare the k8s nmstate nncp to apply
+  vars:
+    nmstate_nncp_manifest_name: "{{ hostvars[item].ansible_hostname }}"
+    nmstate_nncp_overrides:
+      metadata:
+        labels:
+          osp/interface: "{{ cifmw_ci_nmstate_network_layout[item].default.iface }}"
+        name: "{{ nmstate_nncp_manifest_name }}"
+      spec:
+        desiredState:
+          interfaces: "{{ _cifmw_ci_nmstate_configs[item].interfaces }}"
+          dns-resolver: "{{ _cifmw_ci_nmstate_configs[item].dns }}"
+          routes: "{{ _cifmw_ci_nmstate_configs[item].routes }}"
+        nodeSelector:
+          kubernetes.io/hostname: "{{ hostvars[item].ansible_hostname }}"
+  ansible.builtin.set_fact:
+    _cifmw_ci_nmstate_k8s_node_config_name: "{{ nmstate_nncp_manifest_name }}"
+    _cifmw_ci_nmstate_k8s_node_config: >-
+      cifmw_ci_nmstate_nncp_config_template |
+      combine(nmstate_nncp_overrides, recursive=True)
+    cacheable: true
+
+- name: Save k8s nmstate nncp manifests as artifacts
+  ansible.builtin.copy:
+    dest: "{{ cifmw_ci_nmstate_manifests_dir }}/{{ item }}-nncp.yaml"
+    content: "{{ _cifmw_ci_nmstate_k8s_node_config | to_nice_yaml }}"
+
+- name: Provision nmstate nncp for the node worker
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    definition: "{{ _cifmw_ci_nmstate_k8s_node_config }}"
+
+- name: Wait for nsmtate nncp to be ready
+  kubernetes.core.k8s_info:
+    api_version: nmstate.io/v1
+    kind: NodeNetworkConfigurationPolicy
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    name: "{{ _cifmw_ci_nmstate_k8s_node_config_name }}"
+  register: _cifmw_ci_nsmate_intance_nncp_out
```

Comment:
ditto - shorter name might be good here, since "internal" use only for that specific task.